packages/loot-core/src/server/budget/goaltemplates.js

```diff
@@ -329,19 +329,25 @@ async function applyCategoryTemplate(category, template_lines, month, force) {
         break;
       }
       case 'percentage': {
-        /*
-          let income_category = (await actual.getCategories()).filter(c => c.is_income == true && c.name == template.category);
-          let func = (getBudgetMonthTestFunc || getBudgetMonth);
-          let budget = await func(month);
-          for (var g = 0; g < budget.categoryGroups.length; g++) {
-            if (income_category.group_id == budget.categoryGroups[g].id) {
-              for (var c = 0; c < budget.categoryGroups[g].categories.length; c++)
-                if (income_category.id == budget.categoryGroups[g].categories[c].id) {
-                  let month_category = budget.categoryGroups[g].categories[c];
-                }
-            }
+        let percent = template.percent;
+        let monthlyIncome = 0;
+        if (template.category === 'all income') {
+          monthlyIncome = await getSheetValue(sheetName, `total-income`);
+        } else {
+          let income_category = (await db.getCategories()).filter(
+            c => c.is_income == true && c.name == template.category,
+          );
+          if (!income_category) {
+            throw new Error(`Could not find category “${template.category}”`);
           }
-          */
+          monthlyIncome = await getSheetValue(
+            sheetName,
+            `sum-amount-${income_category[0].id}`,
+          );
+        }
+        if (monthlyIncome > 0) {
+          to_budget = Math.round(monthlyIncome * (percent / 100));
+        } else to_budget = 0;
```

Comment:
Suggestion: either replace this with a call to `Math.max`
```suggestion
        to_budget = Math.max(0, Math.round(monthlyIncome * (percent / 100)));
```

…or add braces to the `else`
```suggestion
        if (monthlyIncome > 0) {
          to_budget = Math.round(monthlyIncome * (percent / 100));
        } else {
          to_budget = 0;
        }
```

Reply:
done.